src/systems/triggered_publisher/TriggeredPublisher.cc

```diff
@@ -643,6 +699,47 @@ void TriggeredPublisher::Configure(const Entity &,
 
   this->workerThread =
       std::thread(std::bind(&TriggeredPublisher::DoWork, this));
+  this->serviceWorkerThread =
+      std::thread(std::bind(&TriggeredPublisher::DoServiceWork, this));
+}
+//////////////////////////////////////////////////
+
+#define HANDLE_REQUEST(requestT, typeString) \
+if (((serviceInfo.reqType == typeString) | serviceInfo.reqType.empty()) \
+  && !isProcessing) \
+{ \
+  this->HandleRequest<requestT>(serviceInfo); \
+  isProcessing = true; \
+} \
+
+void TriggeredPublisher::DoServiceWork()
+{
+  while (!this->srvDone)
+  {
+    {
+      using namespace std::chrono_literals;
+      std::unique_lock<std::mutex> lock(this->serviceCountMutex);
+      this->serviceMatchSignal.wait(lock,
+        [this]
+        {
+          return (this->callService) || this->srvDone;
+        });
+
+      if (this->srvDone)
+      {
+        continue;
+      }
+    }
+    for (auto &serviceInfo : this->serviceOutputInfo)
+    {
+      bool isProcessing {false};
+      HANDLE_REQUEST(msgs::Pose, "ignition.msgs.Pose");
+      HANDLE_REQUEST(msgs::StringMsg, "ignition.msgs.StringMsg");
+      HANDLE_REQUEST(msgs::Boolean, "ignition.msgs.Boolean");
+      HANDLE_REQUEST(msgs::Empty, "ignition.msgs.Empty");
```

Comment:
If you take my suggestion above, then this will end up being something like:

```
handle_request<msgs::Pose>(serviceInfo, "ignition.msgs.Pose");
handle_request<msgs::StringMsg>(serviceInfo, "ignition.msgs.StringMsg");
handle_request<msgs::Boolean>(serviceInfo, "ignition.msgs.Boolean");
handle_request<msgs::Empty>(serviceInfo, "igmition.msgs.Empty");
```